### PR TITLE
Local storage

### PR DESCRIPTION
**Pull request by peterbertel**
The pull request introduces the initial implementation of the local storage. Each new Day will be saved to a backend database json file. The contents of this file will persist until the app is deleted from the user's device. The new BackendService invokes each of these actions for submitting data to and retrieving data from the database.

This pull request will also update the new_summary.dart page, which will create new Day objects based on the contents in the backend database file.

For testing purposes, add several new days by refreshing the app after each successful entry. In a future update, we will update the logic to only allow one entry per day.

### CODE REVIEW

File: lib/pages/questions/questions_controller.dart [hunk 1]
```diff
@@ -35,24 +38,25 @@ class _QuestionsControllerState extends State<QuestionsController> {
...
 
     CustomContinueButton continueButton = CustomContinueButton(
       onPressed: () async {
-        day.date = DateTime.now();
-        print(day.date);
-        print(day.date.toUtc());
-        print(day.date.toLocal());
-        print(day.date.toUtc().toLocal());
-        await dayService.addDocument(day.toJson());
+        // await dayService.addDocument(day.toJson());
```
Comment (ewynx) [inline lib/pages/questions/questions_controller.dart]: what should happen with this line?

File: lib/pages/summary/new_summary.dart [hunk 2]
```diff
@@ -28,20 +31,22 @@ class NewSummary extends StatelessWidget {
...
 
-    Day day1 = Day(
-        id: "first",
-        date: DateTime.now(),
-        socialDistance: true,
-        feelings: feels.take(6).toList().toSet(),
-        activities: acts.take(5).toList().toSet(),
-        note:
-            'Today was a pretty good day! I read a book and went for a run by myself.');
+    // Day day1 = Day(
```
Comment (ewynx) [inline lib/pages/summary/new_summary.dart]: can be removed?

File: lib/pages/summary/new_summary.dart [hunk 3]
```diff
@@ -126,6 +131,10 @@ class NewSummary extends StatelessWidget {
         activities: acts.take(2).toList().toSet(),
         note: 'This is the second day');
 
+    // List<Day> days = [day1, day2, day3, day4, day5, day6, day7, day8, day9, day10, day11];
```
Comment (ewynx) [inline lib/pages/summary/new_summary.dart]: dito :D

File: lib/services/backend_service.dart [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import 'package:shelter_in_place/models/day_model.dart';
+import 'package:path_provider/path_provider.dart';
+
+class BackendService {
+
+  final databaseFilename = 'distince-day-entries.json';
+
+  Future<String> getDatabasePath() async {
+    final directory = await getApplicationDocumentsDirectory();
+    // final directory = await getTemporaryDirectory();
```
Comment (ewynx) [inline lib/services/backend_service.dart]: can this be removed?

**Review by ewynx — CHANGES_REQUESTED** [review]
Works like a charm!! Awesome 👍 
I added some comments, it's probably code that can all be removed, so just for the sake of cleanness :)

File: lib/pages/summary/new_summary.dart (hunk 2 above)
Comment (ewynx) [reply lib/pages/summary/new_summary.dart]: actually the definition of all days day1, day2 etc can be removed!

File: lib/services/backend_service.dart (hunk 4 above)
Comment (peterbertel) [reply lib/services/backend_service.dart]: Yep, I removed it in [this commit](https://github.com/ewynx/shelter_in_place/pull/24/commits/53ca9c64c536a774cb80555dbeaf4d3205193c59).

File: lib/pages/summary/new_summary.dart (hunk 3 above)
Comment (peterbertel) [reply lib/pages/summary/new_summary.dart]: Deleted in [this commit](https://github.com/ewynx/shelter_in_place/pull/24/commits/53ca9c64c536a774cb80555dbeaf4d3205193c59).

File: lib/pages/summary/new_summary.dart (hunk 2 above)
Comment (peterbertel) [reply lib/pages/summary/new_summary.dart]: I deleted all the unused Day objects in [this commit](https://github.com/ewynx/shelter_in_place/pull/24/commits/53ca9c64c536a774cb80555dbeaf4d3205193c59).

File: lib/pages/questions/questions_controller.dart (hunk 1 above)
Comment (peterbertel) [reply lib/pages/questions/questions_controller.dart]: It wasn't providing any extra functionality for the backend storage, so I had originally just commented it out. I think we can safely delete it, so I did in [this commit](https://github.com/ewynx/shelter_in_place/pull/24/commits/53ca9c64c536a774cb80555dbeaf4d3205193c59).

**Comment by peterbertel** [comment]
@ewynx I deleted all the unnecessary code you referenced in your comments, so I am going to merge in the changes.

**Pull request merged**